src/composables/info/feature-info.composable.ts

```diff
@@ -0,0 +1,413 @@
+      }
+    })
+
+    watchEffect(() => {
+      if (fid.value) {
+        ;(async () => {
+          await getFeatureInfoById(fid.value as string)
+        })()
+      }
+    })
+  }
+
+  async function getFeatureInfoById(fid: string): Promise<void> {
+    const fids = fid.split(',')
+    for (const curFid of fids) {
+      const splittedFid = curFid.split('_')
+      const layersList = [splittedFid[0]]
+      const layerLabel: { [key: string]: string } = {}
+
+      try {
+        if (layersList.length > 0) {
+          setLoading(true)
+          map.getViewport().style.cursor = 'wait'
+
+          const url = new URL(INFO_SERVICE_URL)
+          url.searchParams.append('fid', curFid)
+          try {
+            const resp = await fetch(url.toString())
+
+            let openInfoPanel = false
+            if (screenSizeIsAtLeast('sm')) {
+              openInfoPanel = true
+              setInfoPanelHidden(false)
+            } else {
+              setInfoPanelHidden(true)
+            }
+
+            const node = findById(splittedFid[0])
+            if (node) {
+              layerLabel[splittedFid[0]] = node.layers as string
+            }
+
+            if (resp.ok) {
+              const data = await resp.json()
+              if (data.length > 0) {
+                showInfo(true, data, layerLabel, openInfoPanel, true)
+              }
+            }
+          } catch (error) {
+            reset()
+          }
+        }
+      } catch (error) {
+        reset()
+      }
+    }
+  }
+
+  async function singleclickEvent(evt: MapBrowserEvent<any>): Promise<void> {
+    const layers = map.getLayers().getArray()
+    const layersList = []
+    const layerLabel: { [key: string]: string } = {}
+
+    for (let i = layers.length - 1; i >= 0; i--) {
+      const metadata = layers[i].get('metadata')
+      if (metadata !== undefined && metadata !== null) {
+        if (
+          metadata['is_queryable'] &&
+          layers[i].getVisible() &&
+          layers[i].getOpacity() > 0
+        ) {
+          const queryableId = layers[i].get('queryable_id')
+          layersList.push(queryableId)
+          layerLabel[queryableId] = layers[i].get('label')
+        }
+      }
+    }
+
+    if (layersList.length > 0) {
+      const resolution = map.getView().getResolution()
+      let bigBuffer = 0
+      let smallBuffer = 0
+      if (resolution) {
+        bigBuffer = 20 * resolution
+        smallBuffer = 1 * resolution
+      }
+
+      const point = transform(
+        evt.coordinate,
+        map.getView().getProjection(),
+        PROJECTION_LUX
+      )
+      const big_box = [
+        [point[0] - bigBuffer, point[1] + bigBuffer],
+        [point[0] + bigBuffer, point[1] - bigBuffer],
+      ]
+      const small_box = [
+        [point[0] - smallBuffer, point[1] + smallBuffer],
+        [point[0] + smallBuffer, point[1] - smallBuffer],
+      ]
+
+      setLoading(true)
+      map.getViewport().style.cursor = 'wait'
```

Comment:
it looks like this can be done in a watcher

Reply:
True, kept it as is for now.